### PR TITLE
Hoops and pegs

### LINKED ISSUE #43: [Hoops and pegs] Hoops and pegs textures
Task for ([link to user story](https://github.com/filipkauffeldt/oopp-grupp22/issues/42))

# Textures
Make images of hoops and pegs

### PR DESCRIPTION
**Pull request by Oskmey**
## Summary

This pull request refactors the `LevelView` class in the `com.crocket` package. The class now implements the `ILevelView` interface and provides methods to draw different types of entities on different layers of a `JLayeredPane`.

## Changes

- Added a `TextureManager` instance to manage textures.
- Implemented the `paintComponent` method to draw surfaces.
- Added methods to draw entities on different layers (`drawLowerLayerEntity`, `drawMiddleLayerEntity`, `drawNoneLayeredEntity`).
- Added a `makeLabel` method to create a `JLabel` for an entity.
- Added a `drawDirectionLine` method (currently empty and marked as TODO).
- Added a `getHoopTexture` method to get the texture for a hoop entity based on its rotation.
- Implemented the `drawEntities` method to draw a set of entities.
- Implemented the `drawEntity` method to draw a single entity based on its type.
- Added a `setSurfaceMap` method to set the surface map and level dimensions.

Closes #43 

## Testing

-View is hard to test :(

## Next Steps

- Implement the `drawDirectionLine` method.
- Optimize the `drawEntities` method to avoid unnecessary redraws.

### CODE REVIEW

File: crocket/src/main/java/com/crocket/Application.java [hunk 1]
```diff
@@ -10,8 +10,9 @@ public class Application {
     private static ILevel Level1 = new Level1();
     private static LevelView level1View = new LevelView();
 
-    public static void main(String[] args) {
-        init();
+    public static void main( String[] args )
+    {
```
Comment (filipkauffeldt) [inline crocket/src/main/java/com/crocket/Application.java:14]: Why did you change this?

File: crocket/src/main/java/com/crocket/CroquetView.java [hunk 2]
```diff
@@ -51,6 +51,7 @@ public void setLevelView(LevelView level) {
     public void setBallToLevel(DrawBall ball){
         try{
             this.ball = ball;
+            ball.setSize(19, 19);
```
Comment (filipkauffeldt) [inline crocket/src/main/java/com/crocket/CroquetView.java:54]: Is this still used?

**Review by filipkauffeldt — APPROVED** [review]

File: crocket/src/main/java/com/crocket/CroquetView.java (hunk 2 above)
Comment (Oskmey) [reply crocket/src/main/java/com/crocket/CroquetView.java:54]: No

File: crocket/src/main/java/com/crocket/Application.java (hunk 1 above)
Comment (Oskmey) [reply crocket/src/main/java/com/crocket/Application.java:14]: I think format document did that

File: crocket/src/main/java/com/crocket/DrawableEntity.java [hunk 3]
```diff
@@ -39,4 +40,10 @@ public int getRotation() {
     public EntityType getType() {
         return type;
     }
+
+    public String getTypeName() {
+        return type.name();
```
Comment (Pontus-Wikstrom) [inline crocket/src/main/java/com/crocket/DrawableEntity.java:45]: Is this one necessary? Why call for EntityType methods in DrawableEntity class?

File: crocket/src/main/java/com/crocket/LevelView.java [hunk 4]
```diff
@@ -1,70 +1,142 @@
...
+
+    private void drawLowerLayerEntity(DrawableEntity entity, BufferedImage bufferedImage) {
+        JLabel label = makeLabel(entity, bufferedImage);
+        add(label, JLayeredPane.DEFAULT_LAYER);
+    }
+
+    private void drawMiddleLayerEntity(DrawableEntity entity, BufferedImage bufferedImage) {
+        JLabel label = makeLabel(entity, bufferedImage);
+        add(label, JLayeredPane.MODAL_LAYER);
+    }
```
Comment (Pontus-Wikstrom) [inline crocket/src/main/java/com/crocket/LevelView.java:56]: What is the purpose of dividing the methods in Middle and Lower, what's the different? Just a question

**Review by Pontus-Wikstrom — APPROVED** [review]
Check comments, otherwise is there any possible tests you can do? Approved if there's no need for tests

File: crocket/src/main/java/com/crocket/DrawableEntity.java (hunk 3 above)
Comment (Oskmey) [reply crocket/src/main/java/com/crocket/DrawableEntity.java:45]: Yes it might seem weird but it is a result of the texture manger class.  Essentially, we utilize a JSON file (JSON does not have enums) to map strings to corresponding textures. The strings share the same names as the enums. To enhance readability and adhere to the Law of Demeter, we use ```getTypeName()``` instead of ```entity.getType().name()```.

File: crocket/src/main/java/com/crocket/LevelView.java (hunk 4 above)
Comment (Oskmey) [reply crocket/src/main/java/com/crocket/LevelView.java:56]: It is just the add method that is different.  I wanted to use a JLayeredPane so that we can draw entities on layers.
